component/core/src/main/java/org/exoplatform/social/core/space/SpaceUtils.java

```diff
@@ -1891,17 +1891,29 @@ public static boolean isUserHasMembershipTypesInGroup(String remoteId, String gr
    * @return boolean true if the user has that role Or the role is not present in the space
    * @throws Exception
    */
-  public static boolean isRedactor (String userName, String spaceGroupId) throws Exception{
-    String REDACTOR_MEMBERSHIP_NAME = "redactor";
-
-    List<MembershipType> membershipTypes = getOrganizationService().getMembershipHandler().findMembershipTypesByGroup(spaceGroupId);
-    for(MembershipType membershipType : membershipTypes) {
-      if(REDACTOR_MEMBERSHIP_NAME.equals(membershipType.getName())) {
-        return getOrganizationService().getMembershipHandler().
-                findMembershipByUserGroupAndType(userName, spaceGroupId, REDACTOR_MEMBERSHIP_NAME) != null;
-      }
+  public static boolean isRedactorOrNotSpaceRedactional(String userName, String spaceGroupId) throws Exception {
+    Space space = getSpaceService().getSpaceByGroupId(spaceGroupId);
+    boolean isRedactorOrNotSpaceRedactional = false;
```

Comment:
```suggestion
    return getSpaceService().isRedactor(space, userName) || space.getRedactors().length == 0
```

We should also change the description:

/**
   * Check if the user has the role redactor in that space or that the space is not redactional
   * @param userName
   * @param spaceGroupId
   * @return boolean true if the user has that redactor role Or the space is not redactional
   * @throws Exception
   */

Reply:
we should keep the name of method, isRedactor because used in many places, we should check if space != null to avoid NPE

Reply:
for space.getRedactors().length never return zero, because when create the space the creator is set as redactor by default,

Reply:
We will not add the creator as redactor for now as the managers will not be displayed in the redactors filter